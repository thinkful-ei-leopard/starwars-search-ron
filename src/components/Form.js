import React, {Component} from 'react';
import Results from './Results';

export default class Form extends Component {

    constructor(props) {
        super(props);

        this.state = {
            inputText: {
                value: '',
                touched: false
            }
        };
    }


    componentDidMount() {

    }

    validateSearchText(){
    // this removes any whitespace from value and makes sure the value isn't an empty string
        const userInput = this.state.inputText.value.trim();
        if(userInput.length === 0) {
            return 'Please enter search term';
        } else if(userInput.length > 35) {
            return 'Please keep your search term under 35 characters. That should be enough for most alien languages.';
        }
    }

    addCharacter(){

    }

    handleSubmit(event){
        event.preventDefault();

    }
    
    render() {
        return(
            <form className="Form__container" onSubmit={e => this.handleSubmit(e)}>
                <h2>Star Wars API!</h2>  
                <div className="form-group">
                <label htmlFor="search-character-name">Search characters name: </label>
                <input 
                    type="text" 
                    className="Form__control"
                    name="search-character-name" 
                    id="search-character-name"
                    // Each time the user types anything into input, we update the state. This allows real-time error messages (ie before submit)
                    onChange={e => this.addCharacter(e.target.value)}
                    required/>
                {/* Conditional rendering depending on whether the user has changed the input or not */}
                {/* {this.state.name.touched && (<ValidationError message={nameError} />)} */}
                </div>

        
                <div className="Form__button__group">
                    <button 
                        type="submit" 
                        className="Form__button"
                        // if anything triggers our validation method, the button gets disabled
                        disabled={this.validateSearchText()}>
                        Save
                    </button>
                </div>
                
            </form>
        );
    }
}

// For Fetch. Search people by this: where r2 is the string you wish to search.
  // https://swapi.co/api/people/?search=r2
  // get input from user
  // validate it
  // send API request to that URL with the string at the end (where it says r2)
  // update state -> it triggers render -> values load in DOM

  // config.API_ENDPOINT =https://swapi.co/api/
  // path = people/
  // query_string = ?search=
  // Users_input = STRING
  // all together: 
  // config.API_ENDPOINT + path + query_string + Users_input === https://swapi.co/api/people/?search=r2