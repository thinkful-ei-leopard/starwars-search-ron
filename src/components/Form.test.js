
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Form from './Form';

describe('<Form />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Form />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<Form />).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});