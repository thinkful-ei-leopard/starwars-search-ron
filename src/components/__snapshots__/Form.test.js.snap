// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders this UI as expected 1`] = `
<form
  className="Form__container"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="search-character-name"
    >
      Search characters name: 
    </label>
    <input
      className="Form__control"
      id="search-character-name"
      name="search-character-name"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </div>
  <div
    className="Form__button__group"
  >
    <button
      className="Form__button"
      disabled="Please enter search term"
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
